syft/workers/websocket_client.py

```diff
@@ -32,20 +36,36 @@ def __init__(
 
         # creates the connection with the server which gets held open until the
         # WebsocketClientWorker is garbage collected.
-        self.ws = create_connection(self.uri)
+        self.ws = websocket.create_connection(self.uri, max_size=None)
 
         super().__init__(hook, id, data, is_client_worker, log_msgs, verbose)
 
-    def _send_msg(self, message: bin) -> bin:
+    def _send_msg(self, message: bin, location) -> bin:
         raise RuntimeError(
             "_send_msg should never get called on a ",
             "WebsocketClientWorker. Did you accidentally "
             "make hook.local_worker a WebsocketClientWorker?",
         )
 
+    def _receive_action(self, message: bin) -> bin:
+        self.ws.send(str(binascii.hexlify(message)))
+        response = binascii.unhexlify(self.ws.recv()[2:-1])
+        return response
+
     def _recv_msg(self, message: bin) -> bin:
         """Forwards a message to the WebsocketServerWorker"""
 
-        self.ws.send(str(binascii.hexlify(message)))
-        response = binascii.unhexlify(self.ws.recv()[2:-1])
+        response = self._receive_action(message)
```

Comment:
This will fail or hang if the websocket is not connected. What is the check after this actually doing?

Reply:
_receive_action will not fail, if the websocket is not connected. but the response will be empty. 
The check on self.ws.connected has to be after a self.ws.recv() call as otherwise the variable is not updated. 
So the steps are: 1) try to send/receive, 2) check if connection is ok, if not 3) reconnect and send/receive again. 4) check if connection is ok now, if not fail.